### ISSUE TITLE
Model query failing while selecting a particular column with limit/pagination 

### ISSUE THREAD

**Issue by nkman** [balderdashy/sails-mongo#301]
Query for selecting field firstname and lastname is :-

```sh
User.find(
	{
		'$and': [
			{
				collegeId: collegeId
			},

			{
				'$or': [
					{
						firstname: new RegExp(name, "i")
					},
					{
						lastname: new RegExp(name, "i")
					}
				]
			}
		]
	},
	{
		firstname: 1,
		lastname: 1
	}).exec(function(err, users){
	if(err){
		sails.log.error(err);
		return cb(err, null);
	}

	return cb(null, users);
});
```
The result is ..
```sh
[
    {
        "firstname": "Nairitya",
        "lastname": "Khilari",
        "id": "558ebd154395dd6033237df0"
    }
]
```

This is right and expected. But when you add ``limit`` or ``pagination`` in the query it returns all fields of the collection. 
For example this query
```sh
User.find(
	{
		'$and': [
			{
				collegeId: collegeId
			},

			{
				'$or': [
					{
						firstname: new RegExp(name, "i")
					},
					{
						lastname: new RegExp(name, "i")
					}
				]
			}
		]
	},
	{
		firstname: 1,
		lastname: 1
	}).limit(10).exec(function(err, users){      //or  }).paginate({page: page, limit: 10}).exec(function(err, users){
	if(err){
		sails.log.error(err);
		return cb(err, null);
	}

	return cb(null, users);
});
```
returns
```sh
[
    {
        "email": "n*****@iitr.ac.in",
        "firstname": "Nairitya",
        "lastname": "Khilari",
        "gcmRegistrationId": "NOT_FOUND",
        "collegeId": "558e87f87fafc89221c30542",
        "image": "https://lh3.googleusercontent.com/-MUnp3JDzzNc/AAAAAAAAAAI/AAAAAAAAALs/9RnSOo7YtVM/photo.jpg",
        "gender": "none",
        "isEmailVerified": false,
        "createdAt": "2015-06-27T15:11:17.626Z",
        "updatedAt": "2015-06-27T15:11:17.627Z",
        "id": "558ebd154395dd6033237df0",
        "password": "$2a$10$Z04acoNHuj5BWL/DKkfWG.9C6UkQl38GSg6NdkQCXf8HTyVXWnys6"
    }
]
```
This is not expected. It should have returned data only with fields ``firstname`` and ``lastname``.

**Comment by harianus** [comment]
> ![](https://lh3.googleusercontent.com/-MUnp3JDzzNc/AAAAAAAAAAI/AAAAAAAAALs/9RnSOo7YtVM/photo.jpg)

Nice picture though :-)

**Comment by devinivy** [comment]
What adapter, adapter version, and waterline version are you using?  I recently just saw this similar comment arise: https://github.com/balderdashy/waterline/issues/394#issuecomment-116746191

But I think we have a regression test for this sort of thing, so I wonder if it's an old issue that's been solved.

**Comment by devinivy** [comment]
Wait a moment, typically waterline only takes one parameter in `find`: https://github.com/balderdashy/waterline-docs/blob/master/queries/query-methods.md#find-criteria-callback-

You seem to be using this similarly to the mongodb interface.  In waterline, `select` is used to perform a projection: https://github.com/balderdashy/waterline-docs/blob/5cc4b64207a8cb8e81c53358c7d6024e26e02147/queries/query-language.md#select

**Comment by sailsbot** [comment]
Thanks for posting, @nkman.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
